### PR TITLE
PR loading on bigger displays

### LINKED ISSUE #174: PRs doesn't load on bigger screens
On [OpenPR](https://staging-status.realdevsquad.com/openPRs) and [StalePR](https://staging-status.realdevsquad.com/stale-pr) pages we have infinite scroll implemented.
As the scroller hits the bottom page loads the next set of PRs.
This doesn't seem to work on bigger screens.

Screenshot 2021-08-03 at 11 07 48 AM

### PR DESCRIPTION
**Pull request by rohandhar1234**
#179 
Closes #174

### CODE REVIEW

**Comment by rohandhar1234** [comment]
Closes #174

**Comment by rohandhar1234** [comment]
Please check

**Review by whyDontI — CHANGES_REQUESTED** [review]
Good work 👍🏽 🥳 
let's make it more robust.
Requesting some enhancements.

**Review by whyDontI — CHANGES_REQUESTED** [review]
Can we also have the CardShimmers fill the screen?

File: src/components/PullRequestList/index.tsx [hunk 1]
```diff
@@ -6,6 +6,14 @@ import CardShimmer from '@/components/Loaders/cardShimmer';
...
 import styles from './PullRequestList.module.scss';
 
+const SCREEN_HEIGHT = globalThis.innerHeight;
+const SCREEN_WIDTH = globalThis.innerWidth;
+const CARD_HEIGHT = 147;
+const NUMBER_OF_CARDS_FOR_MOBILE_DISPLAYS = 1;
+const NUMBER_OF_CARDS_FOR_DESKTOP = 3;
+const SMALL_SCREEN_WIDTH_THRESHOLD = 700;
+const HORIZONTAL_NUMBER_OF_CARDS = (SCREEN_WIDTH < SMALL_SCREEN_WIDTH_THRESHOLD
+  ? NUMBER_OF_CARDS_FOR_MOBILE_DISPLAYS : NUMBER_OF_CARDS_FOR_DESKTOP);
```
Comment (whyDontI) [inline src/components/PullRequestList/index.tsx:16]: This approach works great for resolutions above 720p and for phone screens.
But we have 2 horizontal cards for a small desktop screen with a resolution of 800 x 600pixels.
In this case, we end up loading more cards than required.

Can we have some generic approach for any screen width and screen height? Can we calculate the number of cards in a row?

**Comment by rohandhar1234** [comment]
> Can we also have the CardShimmers fill the screen?

I did not understand this. Please elaborate

File: src/components/PullRequestList/index.tsx (hunk 1 above)
Comment (rohandhar1234) [reply src/components/PullRequestList/index.tsx:16]: Yes, right. Could you please suggest any approach or anything that can help me with this

Comment (rohandhar1234) [reply src/components/PullRequestList/index.tsx:16]: I have found a solution to this problem. I will update the PR and then you can check it

File: src/components/PullRequestList/index.tsx [hunk 2]
```diff
@@ -6,6 +6,19 @@ import CardShimmer from '@/components/Loaders/cardShimmer';
 import useFetch from '@/hooks/useFetch';
 import styles from './PullRequestList.module.scss';
 
+const SCREEN_HEIGHT = globalThis.innerHeight;
+const SCREEN_WIDTH = globalThis.innerWidth;
+const CARD_HEIGHT = 147;
+const MINIMUM_CARD_WIDTH = 300;
+const MAXIMUM_NUMBER_OF_CARDS_HORIZONTALLY = 3;
+const MINIMUM_NUMBER_OF_CARDS_HORIZONTALLY = 1;
```
Comment (whyDontI) [inline src/components/PullRequestList/index.tsx:14]: If you look at the CSS styles written for cards in [PullRequestList.module.scss](https://github.com/Real-Dev-Squad/website-status/blob/d852daf809f0fde678228bac369e38243869e72d/src/components/PullRequestList/PullRequestList.module.scss#L33-L36) we can see that the Card's width is `300px` OR `25%` of the `viewport width`.

Using that data we can simplify the logic like this,

```JS
const minCardWidth = 300;
const cardWidth = (
        (screenWidth / 4) > minCardWidth
    )
    ? (screenWidth / 4)
    : minCardWidth;
```

File: src/components/PullRequestList/index.tsx [hunk 3]
```diff
@@ -6,6 +6,19 @@ import CardShimmer from '@/components/Loaders/cardShimmer';
...
+const CALCULATED_NUMBER_OF_CARDS_HORIZONTALLY = Math.floor(SCREEN_WIDTH / MINIMUM_CARD_WIDTH)
+> MINIMUM_NUMBER_OF_CARDS_HORIZONTALLY
+  ? Math.floor(SCREEN_WIDTH / MINIMUM_CARD_WIDTH) : MINIMUM_NUMBER_OF_CARDS_HORIZONTALLY;
+const HORIZONTAL_NUMBER_OF_CARDS = (MAXIMUM_NUMBER_OF_CARDS_HORIZONTALLY
+> CALCULATED_NUMBER_OF_CARDS_HORIZONTALLY
+  ? CALCULATED_NUMBER_OF_CARDS_HORIZONTALLY : MAXIMUM_NUMBER_OF_CARDS_HORIZONTALLY);
+
```
[the remark this answers is not in the document]
Comment (whyDontI) [reply src/components/PullRequestList/index.tsx:21]: Also, convert the variables in `camelCase` after moving them to the function

File: src/components/PullRequestList/index.tsx [hunk 4]
```diff
@@ -36,8 +49,8 @@ const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
   const [noData, setNoData] = useState(false);
   const [page, setPage] = useState(1);
   const [isBottom, setIsBottom] = useState(false);
-
-  const prUrl = `${process.env.NEXT_PUBLIC_BASE_URL}/pullrequests/${prType}?page=${page}`;
+  const numberOfCards = Math.floor((SCREEN_HEIGHT / CARD_HEIGHT) * HORIZONTAL_NUMBER_OF_CARDS);
```
Comment (whyDontI) [inline src/components/PullRequestList/index.tsx:52]: use the function here

**Review by whyDontI — CHANGES_REQUESTED** [review]
Good work @rohandhar1234  👍🏽 
Requesting some more changes, Let me know if you need more details.

**Comment by rohan09-raj** [comment]
> Good work @rohan09-raj 👍🏽
> Requesting some more changes, Let me know if you need more details.

I guess you have tagged wrong rohan😅 @rohandhar1234  is working on this issue.

**Comment by rohandhar1234** [comment]
Please have a look at the changes

- I have made a function that will give us the number of cards and have also converted variables to camelCase as specified.
- I have used minCardsHorizontally to give us correct number of cards for very small displays where screen width is smaller than that of card width
- Since there are only 3 max cards that can be aligned horizontally, I have used the maxCardsHorizontally in the conditional operator and if cards exceed more than 3, we are returned the correct value. eg
 In 1920 x 1080 displays, we will get card width = 480 and number of cards = 4 , but only 3 cards are to be displayed in one row.

File: src/components/PullRequestList/index.tsx (hunk 2 above)
Comment (rohandhar1234) [reply src/components/PullRequestList/index.tsx]: I did not understand the use of getting cardWidth when we can get the value(numberofcards) without calculating it (using the min-cardwidth). I have used this method that you told in the PR that I just updated but I think I might have missed out on what you are trying to convey. Please check

**Review by whyDontI — APPROVED** [review]
Good work @rohandhar1234 🥳

**Pull request merged**